data/lua/wml/harm_unit.lua

```diff
@@ -37,198 +37,220 @@ function wml_actions.harm_unit(cfg)
 			local resistance_multiplier = tonumber(cfg.resistance_multiplier) or 1
 			if harmer_filter then harmer = wesnoth.units.find_on_map(harmer_filter)[1] end
 			-- end of block to support $this_unit
-
-			if animate then
-				if animate ~= "defender" and harmer and harmer.valid then
-					wesnoth.interface.scroll_to_hex(harmer.x, harmer.y, true)
-					wml_actions.animate_unit {
-						flag = "attack",
-						hits = true,
-						with_bars = true,
-						T.filter { id = harmer.id },
-						T.primary_attack ( primary_attack ),
-						T.secondary_attack ( secondary_attack ),
-						T.facing { x = unit_to_harm.x, y = unit_to_harm.y },
-					}
+			
+			if fire_event then
+				local harmer_loc={0, 0}
+				local unit_to_harm_loc={ x = tonumber(unit_to_harm.x) or 0, y = tonumber(unit_to_harm.y) or 0 }
+				if harmer and harmer.valid then
+					harmer_loc = { x = tonumber(harmer.x) or 0, y = tonumber(harmer.y) or 0 }
 				end
-				wesnoth.interface.scroll_to_hex(unit_to_harm.x, unit_to_harm.y, true)
+				wesnoth.game_events.fire("pre harm", unit_to_harm_loc, harmer_loc, primary_attack, secondary_attack)
```

Comment:
This is incorrect, the attacks need to be enclosed in a WML table: `{primary_attack = primary_attack, secondary_attack = secondary_attack}`

Reply:
Like that?
`wesnoth.game_events.fire("pre harm", unit_to_harm_loc, harmer_loc, {primary_attack = primary_attack, secondary_attack = secondary_attack, amount=amount})`
What would damage_inflicted value? amount for pre harm and damage for post harm?
https://github.com/wesnoth/wesnoth/blob/944ed026a255e996a571ee5bbc357f7d1b5d961f/data/lua/wml/harm_unit.lua#L88C4-L94C5